cmd/bootstrap/run/key_generation.go

```diff
@@ -54,3 +58,97 @@ func GenerateKeys(algo crypto.SigningAlgorithm, n int, seeds [][]byte) ([]crypto
 
 	return keys, nil
 }
+
+// WriteJSONFileFunc is a function which writes a file during bootstrapping. It
+// accepts the path for the file (relative to the bootstrapping root directory)
+// and the value to write. The function must marshal the value as JSON and write
+// the result to the given path.
+type WriteJSONFileFunc func(relativePath string, value interface{}) error
+
+// WriteMachineAccountFiles writes machine account key files for a set of nodeInfos.
+// Assumes that machine accounts have been created using the default execution state
+// bootstrapping.
+func WriteMachineAccountFiles(chainID flow.ChainID, nodeInfos []bootstrap.NodeInfo, write WriteJSONFileFunc) error {
+
+	// write machine account key files for each node which has a machine account (LN/SN)
+	//
+	// for the machine account key, we keep track of the address index to map
+	// the Flow address of the machine account to the key.
+	addressIndex := uint64(4)
+	for _, nodeInfo := range nodeInfos {
+
+		// retrieve private representation of the node
+		private, err := nodeInfo.Private()
+		if err != nil {
+			return err
+		}
+
+		// We use the network key for the machine account. Normally it would be
```

Comment:
😕  so the machine account uses the networking key?

Reply:
Yes, but only for test networks.